test/functional/role_test.rb

```diff
@@ -0,0 +1,72 @@
+require File.join(File.dirname(__FILE__), 'test_helper')
+
+describe 'role' do
+  describe 'clone' do
+    before do
+      @cmd = %w(role clone)
+    end
+
+    it 'should print error on missing --id' do
+      params = ['--new-name=zzz']
+
+      expected_result = CommandExpectation.new
+      expected_result.expected_err =
+        ['Could not clone the user role:',
+         "  Missing arguments for 'id'",
+         ''].join("\n")
+      expected_result.expected_exit_code = HammerCLI::EX_USAGE
+
+      api_expects_no_call
+
+      result = run_cmd(@cmd + params)
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should print error on missing --new-name' do
+      params = ['--id=1']
+
+      expected_result = usage_error_result(
+        @cmd,
+        'Option --new-name is required',
+        'Could not clone the user role')
+
+      api_expects_no_call
+
+      result = run_cmd(@cmd + params)
+
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should clone a role by id' do
+      params = ['--id=1', '--new-name=zzz']
+
+      api_expects(:roles, :clone, 'Clone role') do |par|
+        par['id'] == '1' && par['new_name'] == 'zzz'
```

Comment:
I noticed that the apidoc exported for the tests is different from what's currently in https://github.com/theforeman/foreman/pull/4267 The server side parameter is just `name`, not `new_name`.

From the tests:
```
[3] pry(#<HammerCLIForeman::Role::CloneCommand>)> resource.action(:clone).params
=> [<Param *id (String)>, <Param *new_name (String)>]
```

We should update the api export and test `par['name'] == 'zzz'` here.